### ISSUE TITLE
Omnisharp can't find .NET SDK when open any C sharp project

### ISSUE THREAD

**Issue by Netjimmy** [OmniSharp/omnisharp-vscode#4037]
Whenever I open any C sharp project no matter via VS code icon or terminal, it shows such error
`It was not possible to find any .NET SDKs.`
I am pretty sure I have SDK installed
<img width="350" alt="Screen Shot 2020-09-03 at 5 55 42 PM" src="https://user-images.githubusercontent.com/15644582/92181389-b614f100-ee0e-11ea-9ea3-3849d8f7e409.png">

I have tried to follow the [wiki ](https://github.com/OmniSharp/omnisharp-vscode/wiki/Troubleshooting:-'The-.NET-Core-SDK-cannot-be-located.'-errors#general-troubleshooting-steps-on-linuxmac). Reinstall C sharp extension, reinstall SDK, reboot computer, export the `/usr/local/share/dotnet/dotnet` in `~/.bashrc`, `~/.profile`, `~/.zshrc`. None of them work. 

## Environment data
`dotnet --info` output:
.NET Core SDK (reflecting any global.json):
 Version:   3.1.401
 Commit:    39d17847db

Runtime Environment:
 OS Name:     Mac OS X
 OS Version:  10.15
 OS Platform: Darwin
 RID:         osx.10.15-x64
 Base Path:   /usr/local/share/dotnet/sdk/3.1.401/

Host (useful for support):
  Version: 3.1.7
  Commit:  fcfdef8d6b

.NET Core SDKs installed:
  3.1.401 [/usr/local/share/dotnet/sdk]

.NET Core runtimes installed:
  Microsoft.AspNetCore.App 3.1.7 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 3.1.7 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]

To install additional .NET Core runtimes or SDKs:
  https://aka.ms/dotnet-download

VS Code version:
Version: 1.48.2

C# Extension version:
v1.23.2

## OmniSharp log
Starting OmniSharp server at 9/3/2020, 5:44:56 PM
    Target: /Users/jimmy/Desktop/cSharp

OmniSharp server started.
    Path: /Users/jimmy/.vscode/extensions/ms-dotnettools.csharp-1.23.2/.omnisharp/1.37.1/run
    PID: 1287

[info]: OmniSharp.Stdio.Host
        Starting OmniSharp on MacOS 10.15.6 (x64)
[info]: OmniSharp.Services.DotNetCliService
        DotNetPath set to dotnet
[info]: OmniSharp.MSBuild.Discovery.MSBuildLocator
        Located 1 MSBuild instance(s)
            1: StandAlone 16.8.0 - "/Users/jimmy/.vscode/extensions/ms-dotnettools.csharp-1.23.2/.omnisharp/1.37.1/omnisharp/.msbuild/Current/Bin"
[info]: OmniSharp.MSBuild.Discovery.MSBuildLocator
        MSBUILD_EXE_PATH environment variable set to '/Users/jimmy/.vscode/extensions/ms-dotnettools.csharp-1.23.2/.omnisharp/1.37.1/omnisharp/.msbuild/Current/Bin/MSBuild.exe'
[info]: OmniSharp.MSBuild.Discovery.MSBuildLocator
        Registered MSBuild instance: StandAlone 16.8.0 - "/Users/jimmy/.vscode/extensions/ms-dotnettools.csharp-1.23.2/.omnisharp/1.37.1/omnisharp/.msbuild/Current/Bin"
            CscToolExe = csc.exe
            MSBuildToolsPath = /Users/jimmy/.vscode/extensions/ms-dotnettools.csharp-1.23.2/.omnisharp/1.37.1/omnisharp/.msbuild/Current/Bin
            CscToolPath = /Users/jimmy/.vscode/extensions/ms-dotnettools.csharp-1.23.2/.omnisharp/1.37.1/omnisharp/.msbuild/Current/Bin/Roslyn
            BypassFrameworkInstallChecks = true
            MSBuildExtensionsPath = /Users/jimmy/.vscode/extensions/ms-dotnettools.csharp-1.23.2/.omnisharp/1.37.1/omnisharp/.msbuild
[info]: OmniSharp.Cake.CakeProjectSystem
        Detecting Cake files in '/Users/jimmy/Desktop/cSharp'.
[info]: OmniSharp.Cake.CakeProjectSystem
        Could not find any Cake files
[info]: OmniSharp.MSBuild.ProjectSystem
        No solution files found in '/Users/jimmy/Desktop/cSharp'
[info]: OmniSharp.MSBuild.ProjectManager
        Queue project update for '/Users/jimmy/Desktop/cSharp/cSharp.csproj'
[info]: OmniSharp.Script.ScriptProjectSystem
        Detecting CSX files in '/Users/jimmy/Desktop/cSharp'.
[info]: OmniSharp.Script.ScriptProjectSystem
        Could not find any CSX files
[info]: OmniSharp.WorkspaceInitializer
        Invoking Workspace Options Provider: OmniSharp.Roslyn.CSharp.Services.CSharpFormattingWorkspaceOptionsProvider, Order: 0
[info]: OmniSharp.MSBuild.ProjectManager
        Loading project: /Users/jimmy/Desktop/cSharp/cSharp.csproj
[info]: OmniSharp.WorkspaceInitializer
        Invoking Workspace Options Provider: OmniSharp.Roslyn.CSharp.Services.Completion.CompletionOptionsProvider, Order: 0
[info]: OmniSharp.WorkspaceInitializer
        Invoking Workspace Options Provider: OmniSharp.Roslyn.CSharp.Services.RenameWorkspaceOptionsProvider, Order: 100
[info]: OmniSharp.WorkspaceInitializer
        Invoking Workspace Options Provider: OmniSharp.Roslyn.CSharp.Services.ImplementTypeWorkspaceOptionsProvider, Order: 110
[info]: OmniSharp.WorkspaceInitializer
        Invoking Workspace Options Provider: OmniSharp.Roslyn.CSharp.Services.BlockStructureWorkspaceOptionsProvider, Order: 140
[info]: OmniSharp.WorkspaceInitializer
        Configuration finished.
[info]: OmniSharp.Stdio.Host
        Omnisharp server running using Stdio at location '/Users/jimmy/Desktop/cSharp' on host 1262.
It was not possible to find any installed .NET SDKs.
Install a .NET SDK from:
  https://aka.ms/dotnet-download
[warn]: OmniSharp.MSBuild.ProjectManager
        Failed to load project file '/Users/jimmy/Desktop/cSharp/cSharp.csproj'.
/Users/jimmy/Desktop/cSharp/cSharp.csproj(1,1)
Microsoft.Build.Exceptions.InvalidProjectFileException: The SDK 'Microsoft.NET.Sdk' specified could not be found.  /Users/jimmy/Desktop/cSharp/cSharp.csproj
  at Microsoft.Build.Shared.ProjectErrorUtilities.ThrowInvalidProject (System.String errorSubCategoryResourceName, Microsoft.Build.Shared.IElementLocation elementLocation, System.String resourceName, System.Object[] args) [0x00040] in <4322d44f99c84aab9075efd37869f964>:0 
  at Microsoft.Build.Shared.ProjectErrorUtilities.VerifyThrowInvalidProject[T1] (System.Boolean condition, System.String errorSubCategoryResourceName, Microsoft.Build.Shared.IElementLocation elementLocation, System.String resourceName, T1 arg0) [0x00003] in <4322d44f99c84aab9075efd37869f964>:0 
  at Microsoft.Build.Shared.ProjectErrorUtilities.ThrowInvalidProject[T1] (Microsoft.Build.Shared.IElementLocation elementLocation, System.String resourceName, T1 arg0) [0x00000] in <4322d44f99c84aab9075efd37869f964>:0 
  at Microsoft.Build.Evaluation.Evaluator`4[P,I,M,D].ExpandAndLoadImportsFromUnescapedImportExpressionConditioned (System.String directoryOfImportingFile, Microsoft.Build.Construction.ProjectImportElement importElement, System.Collections.Generic.List`1[Microsoft.Build.Construction.ProjectRootElement]& projects, Microsoft.Build.BackEnd.SdkResolution.SdkResult& sdkResult, System.Boolean throwOnFileNotExistsError) [0x00309] in <4322d44f99c84aab9075efd37869f964>:0 
  at Microsoft.Build.Evaluation.Evaluator`4[P,I,M,D].ExpandAndLoadImports (System.String directoryOfImportingFile, Microsoft.Build.Construction.ProjectImportElement importElement, Microsoft.Build.BackEnd.SdkResolution.SdkResult& sdkResult) [0x00027] in <4322d44f99c84aab9075efd37869f964>:0 
  at Microsoft.Build.Evaluation.Evaluator`4[P,I,M,D].EvaluateImportElement (System.String directoryOfImportingFile, Microsoft.Build.Construction.ProjectImportElement importElement) [0x0000d] in <4322d44f99c84aab9075efd37869f964>:0 
  at Microsoft.Build.Evaluation.Evaluator`4[P,I,M,D].PerformDepthFirstPass (Microsoft.Build.Construction.ProjectRootElement currentProjectOrImport) [0x000e6] in <4322d44f99c84aab9075efd37869f964>:0 
  at Microsoft.Build.Evaluation.Evaluator`4[P,I,M,D].Evaluate () [0x0013c] in <4322d44f99c84aab9075efd37869f964>:0 
  at Microsoft.Build.Evaluation.Evaluator`4[P,I,M,D].Evaluate (Microsoft.Build.Evaluation.IEvaluatorData`4[P,I,M,D] data, Microsoft.Build.Construction.ProjectRootElement root, Microsoft.Build.Evaluation.ProjectLoadSettings loadSettings, System.Int32 maxNodeCount, Microsoft.Build.Collections.PropertyDictionary`1[T] environmentProperties, Microsoft.Build.BackEnd.Logging.ILoggingService loggingService, Microsoft.Build.Evaluation.IItemFactory`2[S,T] itemFactory, Microsoft.Build.Evaluation.IToolsetProvider toolsetProvider, Microsoft.Build.Evaluation.ProjectRootElementCacheBase projectRootElementCache, Microsoft.Build.Framework.BuildEventContext buildEventContext, Microsoft.Build.BackEnd.SdkResolution.ISdkResolverService sdkResolverService, System.Int32 submissionId, Microsoft.Build.Evaluation.Context.EvaluationContext evaluationContext, System.Boolean interactive) [0x00047] in <4322d44f99c84aab9075efd37869f964>:0 
  at Microsoft.Build.Evaluation.Project+ProjectImpl.Reevaluate (Microsoft.Build.BackEnd.Logging.ILoggingService loggingServiceForEvaluation, Microsoft.Build.Evaluation.ProjectLoadSettings loadSettings, Microsoft.Build.Evaluation.Context.EvaluationContext evaluationContext) [0x00063] in <4322d44f99c84aab9075efd37869f964>:0 
  at Microsoft.Build.Evaluation.Project+ProjectImpl.ReevaluateIfNecessary (Microsoft.Build.BackEnd.Logging.ILoggingService loggingServiceForEvaluation, Microsoft.Build.Evaluation.ProjectLoadSettings loadSettings, Microsoft.Build.Evaluation.Context.EvaluationContext evaluationContext) [0x00035] in <4322d44f99c84aab9075efd37869f964>:0 
  at Microsoft.Build.Evaluation.Project+ProjectImpl.ReevaluateIfNecessary (Microsoft.Build.BackEnd.Logging.ILoggingService loggingServiceForEvaluation, Microsoft.Build.Evaluation.Context.EvaluationContext evaluationContext) [0x00000] in <4322d44f99c84aab9075efd37869f964>:0 
  at Microsoft.Build.Evaluation.Project+ProjectImpl.ReevaluateIfNecessary (Microsoft.Build.Evaluation.Context.EvaluationContext evaluationContext) [0x00007] in <4322d44f99c84aab9075efd37869f964>:0 
  at Microsoft.Build.Evaluation.Project+ProjectImpl.Initialize (System.Collections.Generic.IDictionary`2[TKey,TValue] globalProperties, System.String toolsVersion, System.String subToolsetVersion, Microsoft.Build.Evaluation.ProjectLoadSettings loadSettings, Microsoft.Build.Evaluation.Context.EvaluationContext evaluationContext) [0x00127] in <4322d44f99c84aab9075efd37869f964>:0 
  at Microsoft.Build.Evaluation.Project..ctor (System.String projectFile, System.Collections.Generic.IDictionary`2[TKey,TValue] globalProperties, System.String toolsVersion, System.String subToolsetVersion, Microsoft.Build.Evaluation.ProjectCollection projectCollection, Microsoft.Build.Evaluation.ProjectLoadSettings loadSettings, Microsoft.Build.Evaluation.Context.EvaluationContext evaluationContext) [0x00073] in <4322d44f99c84aab9075efd37869f964>:0 
  at Microsoft.Build.Evaluation.Project..ctor (System.String projectFile, System.Collections.Generic.IDictionary`2[TKey,TValue] globalProperties, System.String toolsVersion, System.String subToolsetVersion, Microsoft.Build.Evaluation.ProjectCollection projectCollection, Microsoft.Build.Evaluation.ProjectLoadSettings loadSettings) [0x00000] in <4322d44f99c84aab9075efd37869f964>:0 
  at Microsoft.Build.Evaluation.Project..ctor (System.String projectFile, System.Collections.Generic.IDictionary`2[TKey,TValue] globalProperties, System.String toolsVersion, Microsoft.Build.Evaluation.ProjectCollection projectCollection, Microsoft.Build.Evaluation.ProjectLoadSettings loadSettings) [0x00000] in <4322d44f99c84aab9075efd37869f964>:0 
  at Microsoft.Build.Evaluation.Project..ctor (System.String projectFile, System.Collections.Generic.IDictionary`2[TKey,TValue] globalProperties, System.String toolsVersion, Microsoft.Build.Evaluation.ProjectCollection projectCollection) [0x00000] in <4322d44f99c84aab9075efd37869f964>:0 
  at Microsoft.Build.Evaluation.ProjectCollection.LoadProject (System.String fileName, System.Collections.Generic.IDictionary`2[TKey,TValue] globalProperties, System.String toolsVersion) [0x000f7] in <4322d44f99c84aab9075efd37869f964>:0 
  at Microsoft.Build.Evaluation.ProjectCollection.LoadProject (System.String fileName, System.String toolsVersion) [0x00000] in <4322d44f99c84aab9075efd37869f964>:0 
  at OmniSharp.MSBuild.ProjectLoader.EvaluateProjectFileCore (System.String filePath, System.Collections.Generic.IReadOnlyDictionary`2[TKey,TValue] projectConfigurationsInSolution) [0x00160] in <7e6cb753aca94549b693616e0e7d1016>:0 
  at OmniSharp.MSBuild.ProjectLoader.BuildProject (System.String filePath, System.Collections.Generic.IReadOnlyDictionary`2[TKey,TValue] configurationsInSolution) [0x0000d] in <7e6cb753aca94549b693616e0e7d1016>:0 
  at OmniSharp.MSBuild.ProjectFile.ProjectFileInfo.Load (System.String filePath, OmniSharp.MSBuild.ProjectIdInfo projectIdInfo, OmniSharp.MSBuild.ProjectLoader loader, System.Guid sessionId, OmniSharp.Services.DotNetInfo dotNetInfo) [0x00023] in <7e6cb753aca94549b693616e0e7d1016>:0 
  at OmniSharp.MSBuild.ProjectManager+<>c__DisplayClass31_0.<LoadProject>b__0 () [0x00000] in <7e6cb753aca94549b693616e0e7d1016>:0 
  at (wrapper delegate-invoke) System.Func`1[System.ValueTuple`3[OmniSharp.MSBuild.ProjectFile.ProjectFileInfo,System.Collections.Immutable.ImmutableArray`1[OmniSharp.MSBuild.Logging.MSBuildDiagnostic],OmniSharp.MSBuild.Notification.ProjectLoadedEventArgs]].invoke_TResult()
  at OmniSharp.MSBuild.ProjectManager.LoadOrReloadProject (System.String projectFilePath, System.Func`1[TResult] loader) [0x0001b] in <7e6cb753aca94549b693616e0e7d1016>:0 

[fail]: OmniSharp.MSBuild.ProjectManager
        Attempted to update project that is not loaded: /Users/jimmy/Desktop/cSharp/cSharp.csproj

## Steps to reproduce
Whenever open C sharp project from icon or terminal

## Expected  behavior
Able to pick such debugger tools

**Comment by bemnlam** [comment]
Somehow I met similar issue. And then this is what I tried:
- Install Sublime Text
- Install Omnisharp for Sublime
- Load your C# project in Sublime --> I can successfully load the project
- Close Sublime and open VSCode

And then suddenly I found that the errors ... gone.

**Comment by despian** [comment]
I have also started having this problem.

If I launch vscode from finder or the terminal it always loads the incorrect path:

```
/Users/matt/.platformio/penv/bin:/Users/matt/.platformio/penv:/Users/matt/.platformio/python3/bin:/usr/bin:/bin:/usr/sbin:/sbin
```

If i launch it from the command line it loads the correct path:

```
/opt/local/share/java/android-sdk-macosx/platform-tools:/opt/local/bin:/opt/local/sbin:/opt/local/bin:/opt/local/sbin:/Users/matt/.nvm/versions/node/v12.18.1/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/usr/local/share/dotnet:/opt/X11/bin:~/.dotnet/tools:/Library/Apple/usr/bin:/Library/Frameworks/Mono.framework/Versions/Current/Commands:/Applications/Xamarin Workbooks.app/Contents/SharedSupport/path-bin:/opt/local/Library/Frameworks/Python.framework/Versions/3.6/bin:/Users/matt/Library/Python/3.6/bin
```

Strangely, if i close the app after launching from the command line and re-launch it via finder or the dock it will now load the correct path (until the next reboot).

As @bemnlam said it definitely seems to be the same issue as microsoft/vscode#83155 . However, that issue was closed without a resolution.

Seems like it has something to do with **zsh** and maybe **oh-my-zsh**. But I've been running the same setup for 3 years or so and have only just started having this issue. Not sure what changed.

@Netjimmy are you using **oh-my-zsh**?

# vscode version
```
Version: 1.49.1
Commit: 58bb7b2331731bf72587010e943852e13e6fd3cf
Date: 2020-09-16T23:21:17.533Z (4 days ago)
Electron: 9.2.1
Chrome: 83.0.4103.122
Node.js: 12.14.1
V8: 8.3.110.13-electron.0
OS: Darwin x64 19.6.0
```

**Comment by dylinmaust** [comment]
@despian I too am facing this same issue after a recent Mac OS update. I am using oh-my-zsh.

**Comment by despian** [comment]
Since this seems to be a vscode problem I've opened a new issue there:

microsoft/vscode#107169

@dylinmaust and @Netjimmy maybe you can go give it a "thumbs up".

**Comment by jmarolf** [comment]
closing as duplicate of https://github.com/microsoft/vscode/issues/107169

**Comment by justinmchase** [comment]
This doesn't appear to be a duplicate of #107169, simply setting the sdk path into the $PATH variable doesn't fix it.

**Comment by despian** [comment]
@justinmchase microsoft/vscode#107169 is likely the root cause and describes an issue where VSCode is not using the correct path. If you are also experiencing this issue then setting the sdk on your path won't fix the problem, since VSCode is not loading the correct path variable.

Try verifying the path used by VSCode as suggested by bemnlam:

> ...
> To verify my guess, I've created added the following task in `task.json`:
> 
> ```json
> {
>     "version": "2.0.0",
>     "tasks": [
>         {
>             "label": "Show me the PATH",
>             "command": "echo $PATH",
>             "type": "shell",
>             "problemMatcher": "$msCompile"
>         }
>     ]
> }
> ```
> 
> When I launched the app by **clicking the app icon in your computer**, and then run the task, I got this:
> 
> ```shell
> > Executing task: echo $PATH <
> 
> /usr/local/bin:/usr/bin:/bin
> 
> Terminal will be reused by tasks, press any key to close it.
> ```
> ...

**Comment by justinmchase** [comment]
The question is, what is what file is it looking for, what is expected to be on the path? I have `$MSBuildSDKsPath` set and I can see in the trace logs that its loading that correctly, when I run `dotnet build` everything builds fine its just omnisharp itself which keeps failing to load something, but I have no idea what.

**Comment by lu4** [comment]
If you are using `deno` or other extensions for other languages that might affect environment disable them or make sure they are not loaded for your project type. Here's the instance of `.vscode/settings.json` that helped me to fix this problem:

```
{
    "deno.lint": false,
    "deno.enable": false,
    "deno.unstable": false,
    "deno.import_intellisense_origins": {
        "https://deno.land": false
    }
}
```

**Comment by justinmchase** [comment]
I managed to resolve this by uninstalling dotnet via brew, breaking down and going to the website and just downloading and running that installer. Basically brew is somehow installing differently and its not properly tested by the omnisharp crew.

**Comment by iamNoah1** [comment]
@justinmchase you are the man. I spent 2 hours figuring out why omnisharp cannot find my sdks

**Comment by hayesb-emmanuel** [comment]
Likely the same problem I am seeing with Ubuntu 20.04 and 20.10 clean installs. The snap utility properly installs VS Code + NET Core 5, I use the instructions on MS website to add a symbolic link for the snap dotnet install. I can do everything via CLI, but when I open a project folder in VS Code it can't see the NET SDK. I'll see if I can do a clean install of dotnet not using the snap store, see if that fixes the issue.

**Comment by rezanikfal** [comment]
Sometimes the Visual Studio warning will still persist even after installing the .NET SDK. This is because a link/shortcut/alias needs to be created between the .NET executable install location and the Visual Studio path.

The executable was most likely installed in `/usr/local/share/dotnet/dotnet`. A symbolic link needs to be created to `usr/local/bin` for Visual Studio.

To create that link assuming the locations above use this in the Terminal:

`$ ln -s /usr/local/share/dotnet/dotnet /usr/local/bin/`

For Ubuntu & Snap use this instead:

`$ sudo ln -s /snap/dotnet-sdk/current/dotnet /usr/local/bin/`

**Comment by martin-shields-sage** [comment]
> Sometimes the Visual Studio warning will still persist even after installing the .NET SDK. This is because a link/shortcut/alias needs to be created between the .NET executable install location and the Visual Studio path.
> 
> The executable was most likely installed in `/usr/local/share/dotnet/dotnet`. A symbolic link needs to be created to `usr/local/bin` for Visual Studio.
> 
> To create that link assuming the locations above use this in the Terminal:
> 
> `$ ln -s /usr/local/share/dotnet/dotnet /usr/local/bin/`
> 
> For Ubuntu & Snap use this instead:
> 
> `$ sudo ln -s /snap/dotnet-sdk/current/dotnet /usr/local/bin/`

I tried this and unfortuantly still have the same issue. I see the link in the usr/local/bin and if i click it is runs dotnet. But if i run vscode i still get the same issue. my only way to resolve this problem at the moment is to open vscode using terminal

**Comment by hayesb-emmanuel** [comment]
I was able to get this to work, but only after doing a clean build. Just using the Ubuntu/Snap step to setup a symbolic link didn't work, even with reinstalling both VS Code and .NET Core. It may have been holding onto some configuration file or settings.

As a clean install in this order ...
1. Clean Ubuntu (20.04 LTS) Install + Updates
2. Installed .NET Core w/ snap and set the snap alias
3. Created symbolic link to /usr/local/bin/
4. Installed VS Code
5. Added C# extension
6. Created a project and opened the folder in VSCode and it worked correctly without using terminal

I'll likely test this out later with Ubuntu 20.10. 

Thanks!

**Comment by mamaque** [comment]
`sudo ln -s /snap/dotnet-sdk/current/dotnet /usr/local/bin/dotnet`
worked for me

**Comment by justinmchase** [comment]
I still am experiencing this in some cases and I think its because I'm running `dotnet watch run` in a docker container and that is compiling the projects and building PDB's in the context of the container the path is `/var/app/bin/.../example.dll` and if I am running the debugger has an option to rewrite that path to local symbols but it seems like omnisharp is maybe not using relative paths and it can just conk out on intellisense. If I disable my docker container and just build locally it seems to work much better.

**Comment by PanWuming** [comment]
> I have the 5.0.103 [/snap/dotnet-sdk/112/sdk] installed (as shown, using snap) on KDE neon (based on Ubuntu 20.04).
> `sudo ln -s /snap/dotnet-sdk/current/dotnet /usr/local/bin/dotnet` and restart Visual Studio Code.
> worked for me.

This also worked for me. My machine is Ubuntu 18.04 and DotNET 5.0. One must and `/dotnet` at the end of `/usr/local/bin` when to create soft link.

**Comment by justinmchase** [comment]
Just to followup on my previous comment, because I am still experiencing this constantly, when _debugging_ there is a setting you can add to your `launch.json` configurations to help with this:

```json
{
  "sourceFileMap": {
    // mapping of source code inside a container to the source code on a host machine
    "/var/app": "${workspaceRoot}"
}
```

I think omnisharp essentially needs to support something like this, where the pdb's contain paths to the source relative to the docker container where they are being built rather than my local dev machine and it needs an alias to translate the paths to files on my local machine for the build to work directly.

**Comment by peterdeme** [comment]
This is what solved the issue to me:

```
sudo ln -s /usr/local/share/dotnet/x64/dotnet /usr/local/bin/dotnet
```

**Comment by PeacefulCosmos** [comment]
> Sometimes the Visual Studio warning will still persist even after installing the .NET SDK. This is because a link/shortcut/alias needs to be created between the .NET executable install location and the Visual Studio path.
> 
> The executable was most likely installed in `/usr/local/share/dotnet/dotnet`. A symbolic link needs to be created to `usr/local/bin` for Visual Studio.
> 
> To create that link assuming the locations above use this in the Terminal:
> 
> `$ ln -s /usr/local/share/dotnet/dotnet /usr/local/bin/`
> 
> For Ubuntu & Snap use this instead:
> 
> `$ sudo ln -s /snap/dotnet-sdk/current/dotnet /usr/local/bin/`

you are god!

**Comment by irvanherz** [comment]
In linux system, moving these variables from `.bashrc` to `/etc/profile` can make it works.
```
export DOTNET_ROOT=$HOME/.dotnet
export PATH=$PATH:$HOME/.dotnet:$HOME/.dotnet/tools
```
It looks like vscode doesn't load variables from `.bashrc`
